### PR TITLE
Adding device serial number and firmware

### PR DESCRIPTION
**Pull request by merll**
This PR implements reading and decoding known parts of the ID response of the thermostats.

It also adds some tests for the status response. These are not directly related to the new ID part, but to some additional features I was going to implement on the status (presets). It was not possible to separate the tests further without creating dependencies between the two branches.

### CODE REVIEW

File: eq3bt/eq3btsmart.py [hunk 1]
```diff
@@ -149,9 +152,22 @@ def handle_notification(self, data):
             parsed = self.parse_schedule(data)
             self._schedule[parsed.day] = parsed
 
+        elif data[0] == PROP_ID_RETURN:
+            parsed = DeviceId.parse(data)
```
[the remark this answers is not in the document]
Comment (merll) [reply eq3bt/eq3btsmart.py]: It is in fact imported with a star import in ``.structures``, same as several other items (e.g. ``Status``). IMHO it only makes sense to change this for all imports at once. This PR is probably not the place to do it.

[the remark this answers is not in the document]
Comment (rytilahti) [reply eq3bt/eq3btsmart.py]: Sounds fine to me!

File: eq3bt/eq3btsmart.py [hunk 2]
```diff
@@ -388,5 +403,11 @@ def max_temp(self):
         return EQ3BT_MAX_TEMP
 
     @property
-    def away_end(self):
-        return self._away_end
```
Comment (rytilahti) [inline eq3bt/eq3btsmart.py]: Was this removal accidental?

Comment (merll) [reply eq3bt/eq3btsmart.py]: No, it was intentional. There is a completely identical declaration on line 236-238. Therefore my IDE flagged this as a duplicate.

**Comment by rytilahti** [comment]
Sorry for the delay, this got somehow lost.. Thanks for the patch, let's get it merged :+1:

**Pull request merged**